mutters-ink-bot/src/test/java/com/rabidgremlin/mutters/bot/ink/TestSessionRestore.java

```diff
@@ -0,0 +1,72 @@
+package com.rabidgremlin.mutters.bot.ink;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.rabidgremlin.mutters.core.Context;
+import com.rabidgremlin.mutters.core.bot.BotException;
+import com.rabidgremlin.mutters.core.bot.BotResponse;
+import com.rabidgremlin.mutters.core.session.Session;
+
+/** Test session restores when story changes (which can happen if sessions are stored outside of the bot for instance in a external cache).
+ * 
+ * Ink tries to gracefully handle story state restores when a story structure has changed. It is supposed to raise warnings and errors that can
+ * be checked when this occurs, but this is not happening. There is also a case where restore causes wrong path to be followed and incorrect output
+ * to be generated. See testSessionRestoreWhenStoryOptionsChange()
+ * 
+ * Current 'hack' is treat any change of story as an incompatible state change and raise a BadInkStoryState. 
+ *
+ */
+public class TestSessionRestore
+{
+	private static SessionRestoreTestBot storyThreeBot;
+	private static SessionRestoreTestBot storyNewBot;
+
+	@BeforeClass
+	public static void setUpBot()
+	{
+		storyThreeBot = new SessionRestoreTestBot(new SessionRestoreTestBotConfiguration("story_three_options.ink.json"));
+		storyNewBot = new SessionRestoreTestBot(new SessionRestoreTestBotConfiguration("story_new_option.ink.json"));
+	}
+
+	@Test
```

Comment:
Can do @Test(expected = BadInkStoryState.class) I believe

Reply:
Nope because it is a wrapped exception and annotation doesn't support those (at least last time i checked) :(

Reply:
Seems to be a way to do it with rule annotations: https://stackoverflow.com/questions/871216/junit-possible-to-expect-a-wrapped-exception#20759785.

Reply:
That's some convoluted code. Might stick with what is there.